### DISCUSSION TITLE
calculate_change_events

### DISCUSSION THREAD

**Discussion by Akuli** [PurpleMyst/sansio-lsp-client#30] [General]
Should we delete `utils.calculate_change_events`? I have never used it, and it's not used internally in sansio_lsp_client. A comment above it also says it's tricky to get right and hard to test.

**Comment by PurpleMyst** [comment]
Was it not introduced because it could be used in porcupine? What are you doing there?

**Reply by Akuli** [reply]
Porcupine change events already contain all the information needed in langserver. The code that makes those events has decent tests. The information is taken directly from Tk as the changes happen, without needing to compare the old content and new content as strings.

Also, `calculate_change_events` is not suitable for large files, because it needs the entire file contents as two big strings. Porcupine uses strings of file contents in a few places, but I don't think that's a good thing, and I might change it eventually.

**Reply by PurpleMyst** [reply]
That's valid. I'd had this as a passing thought, how most editors don't store their text as a big, huge, string but as something like a rope.
We can remove it if you don't need it.
